src/init.cpp

```diff
@@ -207,6 +207,7 @@ void Shutdown(NodeContext& node)
 
     // After everything has been shut down, but before things get flushed, stop the
     // CScheduler/checkqueue threadGroup
+    if (node.scheduler) node.scheduler->stop();
```

Comment:
Are these guaranteed to fully execute in-order? Can an interruption-point hit while the scheduler is still tearing down?

Reply:
Not sure what you're asking; `node.scheduler->stop()` just sets a  boolean to say "exit the loop" and notifies everything waiting on the condition var to prevent it staying asleep, all the tear down happens in the thread. The interruption could certainly happen while it's tearing down, `stop(); interrupt_all()` happen in one thread, and only afterwards does the other thread even wake up. I don't think boost interrupt will have an effect anymore though, because serviceQueue doesn't use any boost primitives and boost interrupt only sets a flag which is checked by other boost things.